cmd/api/src/test/integration/harnesses.go

```diff
@@ -5504,6 +5504,745 @@ func (s *RootADHarness) Setup(graphTestContext *GraphTestContext) {
 	graphTestContext.NewRelationship(s.ActiveDirectoryGPO, s.ActiveDirectoryOU, ad.GPLink)
 }
 
+type ESC4Template1 struct {
+	CertTemplate1 *graph.Node
+	CertTemplate2 *graph.Node
+	CertTemplate3 *graph.Node
+	CertTemplate4 *graph.Node
+
+	Domain       *graph.Node
+	EnterpriseCA *graph.Node
+
+	NTAuthStore *graph.Node
+	RootCA      *graph.Node
+
+	Group0  *graph.Node
+	Group11 *graph.Node
+	Group12 *graph.Node
+	Group13 *graph.Node
+	Group14 *graph.Node
+	Group15 *graph.Node
+	Group21 *graph.Node
+	Group22 *graph.Node
+	Group23 *graph.Node
+	Group24 *graph.Node
+	Group25 *graph.Node
+	Group31 *graph.Node
+	Group32 *graph.Node
+	Group33 *graph.Node
+	Group34 *graph.Node
+	Group35 *graph.Node
+	Group41 *graph.Node
+	Group42 *graph.Node
+	Group43 *graph.Node
+	Group44 *graph.Node
+	Group45 *graph.Node
+}
+
+func (s *ESC4Template1) Setup(graphTestContext *GraphTestContext) {
+	domainSid := RandomDomainSID()
+	s.CertTemplate1 = graphTestContext.NewActiveDirectoryCertTemplate("CertTemplate1", domainSid, CertTemplateData{
+		ApplicationPolicies:     []string{},
+		AuthenticationEnabled:   true,
+		AuthorizedSignatures:    1,
+		EKUS:                    []string{},
+		EnrolleeSuppliesSubject: true,
+		NoSecurityExtension:     false,
+		RequiresManagerApproval: false,
+		SchemaVersion:           1,
+		SubjectAltRequireSPN:    false,
+		SubjectAltRequireUPN:    false,
+	})
+	s.CertTemplate2 = graphTestContext.NewActiveDirectoryCertTemplate("CertTemplate2", domainSid, CertTemplateData{
+		ApplicationPolicies:     []string{},
+		AuthenticationEnabled:   false,
+		AuthorizedSignatures:    1,
+		EKUS:                    []string{},
+		EnrolleeSuppliesSubject: true,
+		NoSecurityExtension:     false,
+		RequiresManagerApproval: false,
+		SchemaVersion:           1,
+		SubjectAltRequireSPN:    false,
+		SubjectAltRequireUPN:    false,
+	})
+	s.CertTemplate3 = graphTestContext.NewActiveDirectoryCertTemplate("CertTemplate3", domainSid, CertTemplateData{
+		ApplicationPolicies:     []string{},
+		AuthenticationEnabled:   true,
+		AuthorizedSignatures:    1,
+		EKUS:                    []string{},
+		EnrolleeSuppliesSubject: true,
+		NoSecurityExtension:     false,
+		RequiresManagerApproval: false,
+		SchemaVersion:           2,
+		SubjectAltRequireSPN:    false,
+		SubjectAltRequireUPN:    false,
+	})
+	s.CertTemplate4 = graphTestContext.NewActiveDirectoryCertTemplate("CertTemplate4", domainSid, CertTemplateData{
+		ApplicationPolicies:     []string{},
+		AuthenticationEnabled:   true,
+		AuthorizedSignatures:    0,
+		EKUS:                    []string{},
+		EnrolleeSuppliesSubject: true,
+		NoSecurityExtension:     false,
+		RequiresManagerApproval: false,
+		SchemaVersion:           2,
+		SubjectAltRequireSPN:    false,
+		SubjectAltRequireUPN:    false,
+	})
+
+	s.Domain = graphTestContext.NewActiveDirectoryDomain("Domain", domainSid, false, true)
+	s.EnterpriseCA = graphTestContext.NewActiveDirectoryEnterpriseCA("EnterpriseCA", domainSid)
```

Comment:
I looked over the design document and didn't see it mentioned so it may not matter but it looks like the diagrams show a property of `IsUserSpecifiesSanEnabled=true` for the enterpriseCA which is not setup in the code for this harness. If the property doesn't have any effects on whether or not an ESC4 edge is created it would be good to remove it from the diagrams or if it does have effects than we should make sure to set it up here in the code and not just in the harness diagram.